examples/apps/color_finder.py

```diff
@@ -0,0 +1,825 @@
+        squared distance between color and color_range, 
+        which is the sum of the squared distances from 
+        the R, G, B values to their respective ranges
+    '''
+    r, g, b = color
+    minR, maxR, minG, maxG, minB, maxB = color_range
+    r_dist_sqr = 0
+    g_dist_sqr = 0
+    b_dist_sqr = 0
+    if r < minR:
+        r_dist_sqr = (minR - r) ** 2
+    elif r > maxR:
+        r_dist_sqr = (maxR - r) ** 2
+    if g < minG:
+        g_dist_sqr = (minG - g) ** 2
+    elif g > maxG:
+        g_dist_sqr = (maxG - g) ** 2
+    if b < minB:
+        b_dist_sqr = (minB - b) ** 2
+    elif b > maxB:
+        b_dist_sqr = (maxB - b) ** 2
+    sum_dist_sqr = r_dist_sqr + g_dist_sqr + b_dist_sqr
+    return sum_dist_sqr
+
+def hsv_color_distance_sqr(color, color_range):
+    '''Determines the squared euclidean distance between color and color_range.
+
+    Note that we normalize h, minH, and maxH so that they also fall between 0 and 1, instead of 0 and 360.
+
+    Args:
+        color (float, float, float): the H, S, V values of the color
+        color_range(float, float, float, float, float, float): the minimum and maximum for H, S, and V for the color range
+
+    Returns:
+        squared distance between color and color_range, 
+        which is the sum of the squared distances from 
+        the H, S, V values to their respective ranges
+    '''
+    h, s, v = color
+    minH, maxH, minS, maxS, minV, maxV = color_range
+    # for item in [s, v, minS, maxS, minV, maxV]:
+    #     item *= 100.0
+    h_dist_sqr = 0
+    s_dist_sqr = 0
+    v_dist_sqr = 0
+    if h < minH:
+        h_dist_sqr = (minH - h) ** 2
+    elif h > maxH:
+        h_dist_sqr = (maxH - h) ** 2
+    if s < minS:
+        s_dist_sqr = (minS - s) ** 2
+    elif s > maxS:
+        s_dist_sqr = (maxS - s) ** 2
+    if v < minV:
+        v_dist_sqr = (minV - v) ** 2
+    elif v > maxV:
+        v_dist_sqr = (maxV - v) ** 2
+    sum_dist_sqr = h_dist_sqr + s_dist_sqr + v_dist_sqr
+    return sum_dist_sqr
+
+def color_balance(image):
+    '''Adjusts the color data of an image so that the average R, G, B values across the entire image end up equal.
+
+    This is called a 'gray-world' algorithm, because the colors
+    with equal R, G, B values fall along the grayscale.
+    https://web.stanford.edu/~sujason/ColorBalancing/grayworld.html
+
+    Args:
+        image (PIL image): the image being color-balanced
+
+    Returns
+        the PIL image with balanced color distribution
+    '''
+
+    image_array = from_pillow_image(image)
+    image_array = image_array.transpose(2, 0, 1).astype(numpy.uint32)
+    average_g = numpy.average(image_array[1])
+    image_array[0] = numpy.minimum(image_array[0] * (average_g / numpy.average(image_array[0])), 255)
+    image_array[2] = numpy.minimum(image_array[2] * (average_g / numpy.average(image_array[2])), 255)
+    return to_pillow_image(image_array.transpose(1, 2, 0).astype(numpy.uint8))
+
+def from_pillow_image(image):
```

Comment:
minor, but naming in the form of "x_to_y" can make it clearer in all cases, and it makes it extra clear if doing e.g. x_to_y().y_to_z() style work. e.g: "image_to_array" and "array_to_image".